pom.xml

```diff
@@ -13,4 +14,105 @@
 		<module>game-server</module>
 		<module>login-server</module>
 	</modules>
-</project>
\ No newline at end of file
+
+	<properties>
+		<maven.compiler.release>21</maven.compiler.release>
+		<maven.test.skip>true</maven.test.skip>
+
+		<project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
+	</properties>
+
+	<dependencyManagement>
+		<dependencies>
+			<dependency>
+				<groupId>com.aionemu</groupId>
+				<artifactId>commons</artifactId>
+				<version>${project.version}</version>
+			</dependency>
+		</dependencies>
+	</dependencyManagement>
+
+	<build>
+		<pluginManagement>
+			<plugins>
+				<plugin>
+					<groupId>org.apache.maven.plugins</groupId>
+					<artifactId>maven-compiler-plugin</artifactId>
+					<version>3.13.0</version>
+					<configuration>
+						<compilerArgs>
+							<arg>-Xlint:all,-preview,-this-escape</arg>
+						</compilerArgs>
+						<debug>true</debug>
+					</configuration>
+				</plugin>
+				<plugin>
+					<groupId>org.apache.maven.plugins</groupId>
+					<artifactId>maven-jar-plugin</artifactId>
+					<version>3.4.2</version>
+					<configuration>
+						<archive>
+							<manifestEntries>
+								<POMVersion>${project.version}</POMVersion>
+								<Revision>${git.commit.id.abbrev}${git.dirty}</Revision>
+								<Branch>${git.branch}</Branch>
+								<Date>${maven.build.timestamp}</Date>
+							</manifestEntries>
+						</archive>
+					</configuration>
+				</plugin>
+				<plugin>
+					<groupId>org.apache.maven.plugins</groupId>
+					<artifactId>maven-assembly-plugin</artifactId>
+					<version>3.7.1</version>
+				</plugin>
+				<plugin>
+					<groupId>org.apache.maven.plugins</groupId>
+					<artifactId>maven-antrun-plugin</artifactId>
+					<version>3.1.0</version>
+				</plugin>
+				<plugin>
+					<groupId>org.apache.maven.plugins</groupId>
+					<artifactId>maven-surefire-plugin</artifactId>
+					<version>3.2.5</version>
+				</plugin>
+			</plugins>
+		</pluginManagement>
+
+		<plugins>
+			<plugin>
+				<groupId>io.github.git-commit-id</groupId>
+				<artifactId>git-commit-id-maven-plugin</artifactId>
+				<version>8.0.2</version>
```

Comment:
version `9.0.1` [would be available](https://mvnrepository.com/artifact/io.github.git-commit-id/git-commit-id-maven-plugin), but I haven't checked the changelog. maybe this could be upgraded separately (+1 for small incremental steps from my side 😉).

Reply:
You can include 9.0.1 in this PR, since we're updating pretty much everything at this point.
9.0.1 works for me and writes the version to the manifest as it should. Major breaking change is that [Maven 3.6.3 is now required](https://github.com/git-commit-id/git-commit-id-maven-plugin/releases/tag/v9.0.0), which has been out for over five years now.

Reply:
done.